src/core/protocol-authorization.ts

```diff
@@ -109,16 +112,63 @@ export class ProtocolAuthorization {
     await ProtocolAuthorization.verifyActionCondition(tenant, incomingMessage, messageStore);
   }
 
+  /**
+   * Performs protocol-based authorization against the incoming RecordsQuery message.
+   * @throws {Error} if authorization fails.
+   */
+  public static async authorizeQuery(
+    tenant: string,
+    incomingMessage: RecordsQuery,
+    messageStore: MessageStore,
+  ): Promise<void> {
+    // validate that required properties exist in query filter
+    const { protocol, protocolPath, contextId } = incomingMessage.message.descriptor.filter;
+    if (protocolPath === undefined) {
+      throw new DwnError(
+        DwnErrorCode.ProtocolAuthorizationQueryFilterMissingRequiredProperties,
+        'Role-authorized queries must include `protocolPath` in the filter'
+      );
+    }
```

Comment:
Should we front load this check which seems to be structural to `Records.Query.parse()`?

Reply:
I'm on the fence about this one. This check is only run if `recordsQuery.authorizationPayload.protocolRole` is present, and we haven't been validating `authorization` payload fields within `parse()` methods. I'd prefer to keep the check here for now since it's related to `authorization`.

Reply:
We do validate `authorization` payload via `validateMessageSignatureIntegrity()` which is called in `parse()` right? 

I do agree it is debatable where the best place is, though given the pattern of structural integrity check originally using JSON schema and later via code, I do fee `validateMessageSignatureIntegrity()` or maybe a new method called within `parse()` is worth considering.

We can punt this but keep track the consideration by adding a comment such as :
```ts
// consider moving this check to structural check to `parse()` or `validateMessageSignatureIntegrity()`
```

Reply:
🙏